### PR TITLE
feat: New spec to get the duplicate machine id

### PR DESCRIPTION
**Pull request by huali027**
feat: New spec to get the duplicate machine id and the corresponding hostnames and parser
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### CODE REVIEW

**Comment by huali027** [comment]
@xiangce Please help to review, thanks.

**Review by xiangce — CHANGES_REQUESTED** [review]
Besides of the following comments, you also need to add the `insights.specs.default.DefaultSpecs.machine_id` to the default manifest defined in the `collect.py`

File: insights/specs/datasources/machine_ids.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        SkipComponent: When the filters does not exist or the machine id is not
+           in the filters or the machine id is not duplicate or any exception
+           occurs.
+    """
+    filters = sorted((get_filters(Specs.duplicate_machine_id)))
+    if not filters or Specs.machine_id not in broker:
+        raise SkipComponent
+    machine_id_obj = broker[Specs.machine_id]
+    if machine_id_obj and machine_id_obj.content and len(machine_id_obj.content) == 1:
+        machine_id = str(machine_id_obj.content[0].strip())
```
[the remark this answers is not in the document]
Comment (huali027) [reply insights/specs/datasources/machine_ids.py:37]: When I test, it's a unicode, so I transfer it here.

**Comment by huali027** [comment]
@xiangce the code is updated based on comments and also returns the hostnames with the new requirement. Please help to review again, thanks.

**Review by xiangce — APPROVED** [review]

**Pull request merged**